### ISSUE TITLE
Database throws errors while setting up frappe_docker over docker-compose

### ISSUE THREAD

**Issue by rorar** [frappe/frappe_docker#1009]
Hello :-)

My goal is to get ERPnext running in production.

## Description of the issue
I'm trying to get ERPnext running on UnRaid over docker-compose (V2.X).
ERPnext runs on a home server.
The database throws me some error messages.

## Context information (for bug reports)
used docker-compose file -  mainly based on pwd.yml (no .env file used)
```
version: "3"

services:
  backend:
    image: frappe/erpnext-worker:v14.9.0
    container_name: backend
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

  configurator:
    image: frappe/erpnext-worker:v14.9.0
    container_name: configurator
    command:
      - configure.py
    environment:
      DB_HOST: db
      DB_PORT: "3306"
      REDIS_CACHE: redis-cache:6379
      REDIS_QUEUE: redis-queue:6379
      REDIS_SOCKETIO: redis-socketio:6379
      SOCKETIO_PORT: "9000"
    volumes:
      - sites:/home/frappe/frappe-bench/sites

  create-site:
    image: frappe/erpnext-worker:v14.9.0
    container_name: create-site
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets
    entrypoint:
      - bash
      - -c
    command:
      - >
        wait-for-it -t 120 db:3306;
        wait-for-it -t 120 redis-cache:6379;
        wait-for-it -t 120 redis-queue:6379;
        wait-for-it -t 120 redis-socketio:6379;
        export start=`date +%s`;
        until [[ -n `grep -hs ^ common_site_config.json | jq -r ".db_host // empty"` ]] && \
          [[ -n `grep -hs ^ common_site_config.json | jq -r ".redis_cache // empty"` ]] && \
          [[ -n `grep -hs ^ common_site_config.json | jq -r ".redis_queue // empty"` ]];
        do
          echo "Waiting for common_site_config.json to be created";
          sleep 5;
          if (( `date +%s`-start > 120 )); then
            echo "could not find common_site_config.json with required keys";
            exit 1
          fi
        done;
        echo "common_site_config.json found";
        bench new-site frontend --admin-password=admin --db-root-password=admin --install-app payments --install-app erpnext --set-default;
  db:
    image: mariadb:10.6
    container_name: db
    healthcheck:
      test: mysqladmin ping -h localhost --password=admin
      interval: 1s
      retries: 15
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - --character-set-server=utf8mb4
      - --collation-server=utf8mb4_unicode_ci
      - --skip-character-set-client-handshake
      - --skip-innodb-read-only-compressed # Temporary fix for MariaDB 10.6
    environment:
      MYSQL_ROOT_PASSWORD: admin
    volumes:
      - db-data:/var/lib/mysql

  frontend:
    image: frappe/erpnext-nginx:v14.9.0
    container_name: frontend
    deploy:
      restart_policy:
        condition: on-failure
    depends_on:
      backend:
        condition: service_started
      websocket:
        condition: service_started
    environment:
      BACKEND: backend:8000
      FRAPPE_SITE_NAME_HEADER: frontend
      SOCKETIO: websocket:9000
      UPSTREAM_REAL_IP_ADDRESS: 127.0.0.1
      UPSTREAM_REAL_IP_HEADER: X-Forwarded-For
      UPSTREAM_REAL_IP_RECURSIVE: "off"
    volumes:
      - sites:/usr/share/nginx/html/sites
      - assets:/usr/share/nginx/html/assets
    ports:
      - "6363:8080"

  queue-default:
    image: frappe/erpnext-worker:v14.9.0
    container_name: queue-default
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - worker
      - --queue
      - default
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

  queue-long:
    image: frappe/erpnext-worker:v14.9.0
    container_name: queue-long
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - worker
      - --queue
      - long
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

  queue-short:
    image: frappe/erpnext-worker:v14.9.0
    container_name: queue-short
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - worker
      - --queue
      - short
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

  redis-queue:
    image: redis:6.2-alpine
    container_name: redis-queue
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - redis-queue-data:/data

  redis-cache:
    image: redis:6.2-alpine
    container_name: redis-cache
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - redis-cache-data:/data

  redis-socketio:
    image: redis:6.2-alpine
    container_name: redis-socketio
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - redis-socketio-data:/data

  scheduler:
    image: frappe/erpnext-worker:v14.9.0
    container_name: scheduler
    deploy:
      restart_policy:
        condition: on-failure
    command:
      - bench
      - schedule
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

  websocket:
    image: frappe/frappe-socketio:v14.17.1
    container_name: websocket
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - sites:/home/frappe/frappe-bench/sites
      - assets:/home/frappe/frappe-bench/sites/assets

volumes:
  assets:
  db-data:
  redis-queue-data:
  redis-cache-data:
  redis-socketio-data:
  sites:
```

## Steps to reproduce the issue

1.  Use the docker-compose file from above
2. Wait for the db part and see the erros

### Observed result
there are some password and user related issues (see stacktraxce / examples)

### Expected result
Password and connection works

### Stacktrace / full error message if available
**Examples:**
```
- Access denied for user 'root'@'localhost' (using password: YES)
- 2022-12-04 17:09:52 3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '192.168.64.9' (This connection closed normally without authentication)
- [Warning] Access denied for user 'frappe'@'192.168.64.9' (using password: NO)
- [Warning] Aborted connection 8 to db: '_5e5899d8398b5f7b' user: '_5e5899d8398b5f7b' host: '192.168.64.9' (Got an error reading communication packets)
```

**Install-Log**
```
Attaching to backend, configurator, create-site, db, frontend, queue-default, queue-long, queue-short, redis-cache, redis-queue, redis-socketio, scheduler, websocket
db              | 2022-12-04 17:09:30+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.6.11+maria~ubu2004 started.
db              | 2022-12-04 17:09:30+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
db              | 2022-12-04 17:09:30+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.6.11+maria~ubu2004 started.
db              | 2022-12-04 17:09:31+00:00 [Note] [Entrypoint]: Initializing database files
configurator exited with code 0
redis-socketio  | 1:C 04 Dec 2022 17:09:32.372 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-socketio  | 1:C 04 Dec 2022 17:09:32.372 # Redis version=6.2.7, bits=64, commit=00000000, modified=0, pid=1, just started
redis-socketio  | 1:C 04 Dec 2022 17:09:32.372 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-socketio  | 1:M 04 Dec 2022 17:09:32.373 * monotonic clock: POSIX clock_gettime
redis-socketio  | 1:M 04 Dec 2022 17:09:32.373 # A key '__redis__compare_helper' was added to Lua globals which is not on the globals allow list nor listed on the deny list.
redis-socketio  | 1:M 04 Dec 2022 17:09:32.373 * Running mode=standalone, port=6379.
redis-socketio  | 1:M 04 Dec 2022 17:09:32.373 # Server initialized
redis-socketio  | 1:M 04 Dec 2022 17:09:32.374 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis-socketio  | 1:M 04 Dec 2022 17:09:32.374 * Ready to accept connections
create-site     | wait-for-it: waiting 120 seconds for db:3306
redis-cache     | 1:C 04 Dec 2022 17:09:33.470 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-cache     | 1:C 04 Dec 2022 17:09:33.471 # Redis version=6.2.7, bits=64, commit=00000000, modified=0, pid=1, just started
redis-cache     | 1:C 04 Dec 2022 17:09:33.471 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-cache     | 1:M 04 Dec 2022 17:09:33.479 * monotonic clock: POSIX clock_gettime
redis-cache     | 1:M 04 Dec 2022 17:09:33.488 # A key '__redis__compare_helper' was added to Lua globals which is not on the globals allow list nor listed on the deny list.
redis-cache     | 1:M 04 Dec 2022 17:09:33.495 * Running mode=standalone, port=6379.
redis-cache     | 1:M 04 Dec 2022 17:09:33.497 # Server initialized
redis-cache     | 1:M 04 Dec 2022 17:09:33.497 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis-cache     | 1:M 04 Dec 2022 17:09:33.498 * Ready to accept connections
redis-queue     | 1:C 04 Dec 2022 17:09:34.433 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis-queue     | 1:C 04 Dec 2022 17:09:34.433 # Redis version=6.2.7, bits=64, commit=00000000, modified=0, pid=1, just started
redis-queue     | 1:C 04 Dec 2022 17:09:34.433 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
redis-queue     | 1:M 04 Dec 2022 17:09:34.434 * monotonic clock: POSIX clock_gettime
redis-queue     | 1:M 04 Dec 2022 17:09:34.435 # A key '__redis__compare_helper' was added to Lua globals which is not on the globals allow list nor listed on the deny list.
redis-queue     | 1:M 04 Dec 2022 17:09:34.435 * Running mode=standalone, port=6379.
redis-queue     | 1:M 04 Dec 2022 17:09:34.435 # Server initialized
redis-queue     | 1:M 04 Dec 2022 17:09:34.435 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
redis-queue     | 1:M 04 Dec 2022 17:09:34.436 * Ready to accept connections
frontend        | /docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
frontend        | /docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
frontend        | /docker-entrypoint.sh: Launching /docker-entrypoint.d/frappe-entrypoint.sh
frontend        | /docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
frontend        | 10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
frontend        | 10-listen-on-ipv6-by-default.sh: info: /etc/nginx/conf.d/default.conf differs from the packaged version
frontend        | /docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
frontend        | 20-envsubst-on-templates.sh: Running envsubst on /etc/nginx/templates/default.conf.template to /etc/nginx/conf.d/default.conf
frontend        | /docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
frontend        | /docker-entrypoint.sh: Configuration complete; ready for start up
frontend        | 2022/12/04 17:09:34 [notice] 1#1: using the "epoll" event method
frontend        | 2022/12/04 17:09:34 [notice] 1#1: nginx/1.23.2
frontend        | 2022/12/04 17:09:34 [notice] 1#1: built by gcc 11.2.1 20220219 (Alpine 11.2.1_git20220219) 
frontend        | 2022/12/04 17:09:34 [notice] 1#1: OS: Linux 5.19.17-Unraid
frontend        | 2022/12/04 17:09:34 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 40960:40960
frontend        | 2022/12/04 17:09:34 [notice] 1#1: start worker processes
frontend        | 2022/12/04 17:09:34 [notice] 1#1: start worker process 36
frontend        | 2022/12/04 17:09:34 [notice] 1#1: start worker process 37
queue-short     | 17:09:36 Worker rq:worker:26a1ac45876540159592a8f3b568d32c.9aae7f8b6219.8.home-frappe-frappe-bench:short: started, version 1.11.1
queue-short     | 17:09:36 Subscribing to channel rq:pubsub:26a1ac45876540159592a8f3b568d32c.9aae7f8b6219.8.home-frappe-frappe-bench:short
queue-short     | 17:09:36 *** Listening on home-frappe-frappe-bench:short...
queue-short     | 17:09:36 Cleaning registries for queue: home-frappe-frappe-bench:short
backend         | [2022-12-04 17:09:37 +0000] [1] [INFO] Starting gunicorn 20.1.0
backend         | [2022-12-04 17:09:37 +0000] [1] [INFO] Listening at: http://0.0.0.0:8000 (1)
backend         | [2022-12-04 17:09:37 +0000] [1] [INFO] Using worker: gthread
backend         | [2022-12-04 17:09:37 +0000] [8] [INFO] Booting worker with pid: 8
backend         | [2022-12-04 17:09:37 +0000] [9] [INFO] Booting worker with pid: 9
queue-long      | 17:09:38 Worker rq:worker:5c13192efa07491493c92b0c70803658.c945ce213bf4.7.home-frappe-frappe-bench:long: started, version 1.11.1
queue-long      | 17:09:38 Subscribing to channel rq:pubsub:5c13192efa07491493c92b0c70803658.c945ce213bf4.7.home-frappe-frappe-bench:long
queue-long      | 17:09:38 *** Listening on home-frappe-frappe-bench:long...
queue-long      | 17:09:38 Cleaning registries for queue: home-frappe-frappe-bench:long
queue-default   | 17:09:38 Worker rq:worker:e3b185de004d42dcbc5c704d448b8a8f.367ce73ddd8f.6.home-frappe-frappe-bench:default: started, version 1.11.1
queue-default   | 17:09:38 Subscribing to channel rq:pubsub:e3b185de004d42dcbc5c704d448b8a8f.367ce73ddd8f.6.home-frappe-frappe-bench:default
queue-default   | 17:09:38 *** Listening on home-frappe-frappe-bench:default...
queue-default   | 17:09:38 Cleaning registries for queue: home-frappe-frappe-bench:default
db              | 
db              | 
db              | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
db              | To do so, start the server, then issue the following command:
db              | 
db              | '/usr/bin/mysql_secure_installation'
db              | 
db              | which will also give you the option of removing the test
db              | databases and anonymous user created by default.  This is
db              | strongly recommended for production servers.
db              | 
db              | See the MariaDB Knowledgebase at https://mariadb.com/kb
db              | 
db              | Please report any problems at https://mariadb.org/jira
db              | 
db              | The latest information about MariaDB is available at https://mariadb.org/.
db              | 
db              | Consider joining MariaDB's strong and vibrant community:
db              | https://mariadb.org/get-involved/
db              | 
db              | 2022-12-04 17:09:46+00:00 [Note] [Entrypoint]: Database files initialized
db              | 2022-12-04 17:09:46+00:00 [Note] [Entrypoint]: Starting temporary server
db              | 2022-12-04 17:09:46+00:00 [Note] [Entrypoint]: Waiting for server startup
db              | 2022-12-04 17:09:46 0 [Note] mariadbd (server 10.6.11-MariaDB-1:10.6.11+maria~ubu2004) starting as process 184 ...
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Number of pools: 1
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Using Linux native AIO
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Completed initialization of buffer pool
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: 128 rollback segments are active.
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Creating shared tablespace for temporary tables
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
db              | 2022-12-04 17:09:46 0 [Note] InnoDB: 10.6.11 started; log sequence number 42120; transaction id 14
db              | 2022-12-04 17:09:47 0 [Note] Plugin 'FEEDBACK' is disabled.
db              | 2022-12-04 17:09:47 0 [Warning] 'user' entry 'root@753eaeb8c30c' ignored in --skip-name-resolve mode.
db              | 2022-12-04 17:09:47 0 [Warning] 'proxies_priv' entry '@% root@753eaeb8c30c' ignored in --skip-name-resolve mode.
db              | 2022-12-04 17:09:47 0 [Note] mariadbd: ready for connections.
db              | Version: '10.6.11-MariaDB-1:10.6.11+maria~ubu2004'  socket: '/run/mysqld/mysqld.sock'  port: 0  mariadb.org binary distribution
db              | 2022-12-04 17:09:47 3 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
db              | 2022-12-04 17:09:47+00:00 [Note] [Entrypoint]: Temporary server started.
db              | 2022-12-04 17:09:48 6 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
db              | 2022-12-04 17:09:49 7 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
db              | 2022-12-04 17:09:50+00:00 [Note] [Entrypoint]: Securing system users (equivalent to running mysql_secure_installation)
db              | 
db              | 2022-12-04 17:09:50+00:00 [Note] [Entrypoint]: Stopping temporary server
db              | 2022-12-04 17:09:50 0 [Note] mariadbd (initiated by: unknown): Normal shutdown
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: FTS optimize thread exiting.
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Starting shutdown...
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Dumping buffer pool(s) to /var/lib/mysql/ib_buffer_pool
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Buffer pool(s) dump completed at 221204 17:09:50
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Shutdown completed; log sequence number 42132; transaction id 15
db              | 2022-12-04 17:09:50 0 [Note] mariadbd: Shutdown complete
db              | 
db              | 2022-12-04 17:09:50+00:00 [Note] [Entrypoint]: Temporary server stopped
db              | 
db              | 2022-12-04 17:09:50+00:00 [Note] [Entrypoint]: MariaDB init process done. Ready for start up.
db              | 
db              | 2022-12-04 17:09:50 0 [Note] mariadbd (server 10.6.11-MariaDB-1:10.6.11+maria~ubu2004) starting as process 1 ...
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Number of pools: 1
db              | 2022-12-04 17:09:50 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Using Linux native AIO
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Initializing buffer pool, total size = 134217728, chunk size = 134217728
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Completed initialization of buffer pool
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: 128 rollback segments are active.
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Creating shared tablespace for temporary tables
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: 10.6.11 started; log sequence number 42132; transaction id 14
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
db              | 2022-12-04 17:09:51 0 [Note] Plugin 'FEEDBACK' is disabled.
db              | 2022-12-04 17:09:51 0 [Note] InnoDB: Buffer pool(s) load completed at 221204 17:09:51
db              | 2022-12-04 17:09:51 0 [Warning] You need to use --log-bin to make --expire-logs-days or --binlog-expire-logs-seconds work.
db              | 2022-12-04 17:09:51 0 [Note] Server socket created on IP: '0.0.0.0'.
db              | 2022-12-04 17:09:51 0 [Note] Server socket created on IP: '::'.
db              | 2022-12-04 17:09:51 0 [Note] mariadbd: ready for connections.
db              | Version: '10.6.11-MariaDB-1:10.6.11+maria~ubu2004'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
db              | 2022-12-04 17:09:52 3 [Warning] Aborted connection 3 to db: 'unconnected' user: 'unauthenticated' host: '192.168.64.9' (This connection closed normally without authentication)
create-site     | wait-for-it: db:3306 is available after 20 seconds
create-site     | wait-for-it: waiting 120 seconds for redis-cache:6379
create-site     | wait-for-it: redis-cache:6379 is available after 0 seconds
create-site     | wait-for-it: waiting 120 seconds for redis-queue:6379
create-site     | wait-for-it: redis-queue:6379 is available after 0 seconds
create-site     | wait-for-it: waiting 120 seconds for redis-socketio:6379
create-site     | wait-for-it: redis-socketio:6379 is available after 0 seconds
create-site     | common_site_config.json found
db              | 2022-12-04 17:09:53 6 [Warning] Access denied for user 'frappe'@'192.168.64.9' (using password: NO)
db              | 2022-12-04 17:09:56 8 [Warning] Aborted connection 8 to db: '_5e5899d8398b5f7b' user: '_5e5899d8398b5f7b' host: '192.168.64.9' (Got an error reading communication packets)
create-site     | 
create-site     | Installing frappe...
Updating DocTypes for frappe        : [========================================] 100%
Updating country info               : [========================================] 100%
create-site     | Updating Dashboard for frappe
create-site     | 
create-site     | Installing payments...
Updating DocTypes for payments      : [========================================] 100%
create-site     | * Installing Payment Custom Fields in Web Form
create-site     | Updating Dashboard for payments
create-site     | App payments already installed
create-site     | 
create-site     | Installing erpnext...
redis-cache     | 1:M 04 Dec 2022 17:13:46.002 * 10000 changes in 60 seconds. Saving...
redis-cache     | 1:M 04 Dec 2022 17:13:46.013 * Background saving started by pid 15
redis-cache     | 15:C 04 Dec 2022 17:13:46.039 * DB saved on disk
redis-cache     | 15:C 04 Dec 2022 17:13:46.039 * RDB: 0 MB of memory used by copy-on-write
redis-cache     | 1:M 04 Dec 2022 17:13:46.114 * Background saving terminated with success
Updating DocTypes for erpnext       : [========================================] 100%
create-site     | Updating customizations for Address
create-site     | Updating customizations for Contact
create-site     | Updating Dashboard for erpnext
create-site     | *** Scheduler is disabled ***
create-site     | Current Site set to frontend
scheduler       | /usr/local/lib/python3.10/functools.py:981: UserWarning: The 'filters_config' hook used to add custom operators is not yet implemented in frappe.db.query engine. Use db_query (frappe.get_list) instead.
scheduler       |   val = self.func(instance)
db              | 2022-12-04 17:14:39 12 [Warning] Aborted connection 12 to db: '_5e5899d8398b5f7b' user: '_5e5899d8398b5f7b' host: '192.168.64.9' (Got an error reading communication packets)
create-site exited with code 0
backend         | /usr/local/lib/python3.10/functools.py:981: UserWarning: The 'filters_config' hook used to add custom operators is not yet implemented in frappe.db.query engine. Use db_query (frappe.get_list) instead.
backend         |   val = self.func(instance)
frontend        | 192.168.178.22 - - [04/Dec/2022:17:17:55 +0000] "POST /api/method/frappe.core.doctype.log_settings.log_settings.has_unseen_error_log HTTP/1.1" 403 450 "http://192.168.178.55:6363/app/home" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36" "-"
redis-cache     | 1:M 04 Dec 2022 17:18:47.024 * 100 changes in 300 seconds. Saving...
redis-cache     | 1:M 04 Dec 2022 17:18:47.024 * Background saving started by pid 16
redis-cache     | 16:C 04 Dec 2022 17:18:47.068 * DB saved on disk
redis-cache     | 16:C 04 Dec 2022 17:18:47.068 * RDB: 1 MB of memory used by copy-on-write
redis-cache     | 1:M 04 Dec 2022 17:18:47.125 * Background saving terminated with success
frontend        | 192.168.178.22 - - [04/Dec/2022:17:27:55 +0000] "POST /api/method/frappe.core.doctype.log_settings.log_settings.has_unseen_error_log HTTP/1.1" 403 435 "http://192.168.178.55:6363/app/home" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36" "-"
queue-short     | 17:36:36 Cleaning registries for queue: home-frappe-frappe-bench:short
queue-long      | 17:36:38 Cleaning registries for queue: home-frappe-frappe-bench:long
queue-default   | 17:36:38 Cleaning registries for queue: home-frappe-frappe-bench:default
frontend        | 192.168.178.22 - - [04/Dec/2022:17:37:55 +0000] "POST /api/method/frappe.core.doctype.log_settings.log_settings.has_unseen_error_log HTTP/1.1" 403 435 "http://192.168.178.55:6363/app/home" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36" "-"
 
```

**Comment by revant** [comment]
I'm not database expert,

found this on Google search https://www.percona.com/blog/2016/05/16/mysql-got-an-error-reading-communication-packet-errors/

I've not faced such error, may be someone else can help.

in case of slow machines, removing healthcheck from db service helped.

**Comment by rorar** [comment]
@revant thanks for looking into this.
the issue could have a link to these issues:
1. https://discuss.erpnext.com/t/pymysql-err-operationalerror-1045-access-denied-for-user-root-localhost-using-password-yes/82542
2. #448 

I tried this:
https://github.com/frappe/frappe_docker/blob/main/docs/troubleshoot.md#fixing-mariadb-issues-after-rebuilding-the-container
Tried the console and docker compose exec 
but with the command
```
mysql -uroot -p123 -hmariadb
```
It says something like "no connection do MariaDB" (not at home atm.)

**Comment by revant** [comment]
make sure mariadb container is running without errors by verifying logs

**Comment by revant** [comment]
If it is not fatal please avoid or fix and send a PR.

mariadb configuration, root password setup takes some time.

**Comment by revant** [comment]
it seems the site got created successfully.

the errors are due to containers trying to connect to mariadb before it's setup. once the config is ready, it creates the site successfully.

closing the issue because it cannot be solved. if you find any solution re-open new issue or send as PR.

**Comment by xbiggyl** [comment]
I am facing the same issue while setting up image: frappe/erpnext:v14.13.0

I waited for a while after creating the mariadb container.

Followed the single server setup instructions.

But when I execute the command:

```
docker compose --project-name erpnext-one exec backend \
  bench new-site [mysitename] --no-mariadb-socket --mariadb-root-password[password set in mariadb env] --install-app erpnext --admin-password [mypassword]
```

I get the same error as the OP:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 109, in <module>
    main()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/bench_helper.py", line 18, in main
    click.Group(commands=commands)(prog_name="bench")
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/commands/site.py", line 74, in new_site
    _new_site(
  File "/home/frappe/frappe-bench/apps/frappe/frappe/installer.py", line 80, in _new_site
    install_db(
  File "/home/frappe/frappe-bench/apps/frappe/frappe/installer.py", line 156, in install_db
    setup_database(force, source_sql, verbose, no_mariadb_socket)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/__init__.py", line 20, in setup_database
    return frappe.database.mariadb.setup_db.setup_database(
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/mariadb/setup_db.py", line 41, in setup_database
    if force or (db_name not in dbman.get_database_list()):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/db_manager.py", line 49, in get_database_list
    return self.db.sql("SHOW DATABASES", pluck=True)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py", line 199, in sql
    self.connect()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/database.py", line 113, in connect
    self._conn = self.get_connection()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/mariadb/database.py", line 98, in get_connection
    conn = self._get_connection()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/mariadb/database.py", line 104, in _get_connection
    return self.create_connection()
  File "/home/frappe/frappe-bench/apps/frappe/frappe/database/mariadb/database.py", line 107, in create_connection
    return pymysql.connect(**self.get_connection_settings())
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/connections.py", line 353, in __init__
    self.connect()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/connections.py", line 633, in connect
    self._request_authentication()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/connections.py", line 907, in _request_authentication
    auth_packet = self._read_packet()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/home/frappe/frappe-bench/env/lib/python3.10/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1045, "Access denied for user 'root'@'172.27.0.8' (using password: YES)")
```

I even tried the following troubleshooting commands:

https://github.com/frappe/frappe_docker/blob/main/docs/troubleshoot.md#fixing-mariadb-issues-after-rebuilding-the-container

But I keep getting the error. The site folder is created is:

/var/lib/docker/volumes/erpnext-one_sites/_data

But no DB entry in mariadb.
